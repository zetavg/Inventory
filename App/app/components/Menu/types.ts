import type { ColorValue } from 'react-native';

/**
 * The state of the action.
 * - off: A constant indicating the menu element is in the “off” state.
 * - on: A constant indicating the menu element is in the “on” state.
 * - mixed: A constant indicating the menu element is in the “mixed” state.
 */
type MenuState = 'off' | 'on' | 'mixed';

export type MenuAction = {
  type?: 'section';
  onPress?: () => void;
  /**
   * The action's title.
   */
  title?: string;
  /**
   * (iOS only) The SF Symbol name of the icon for the action.
   * @platform iOS
   */
  sfSymbolName?: string;
  // /**
  //  * (Android only)
  //  * The action's title color.
  //  * @platform Android
  //  */
  // titleColor?: number | ColorValue;
  // /**
  //  * (iOS14+ only)
  //  * An elaborated title that explains the purpose of the action.
  //  * @platform iOS
  //  */
  // subtitle?: string;
  /**
   * The attributes indicating the style of the action.
   */
  attributes?: MenuAttributes;
  /**
   * The state of the action.
   */
  state?: MenuState;
  /**
   * An attribute indicating the destructive style.
   */
  destructive?: boolean;
  // /**
  //  * (Android and iOS13+ only)
  //  * - The action's image.
  //  * - Allows icon name included in project or system (Android) resources drawables and
  //  * in SF Symbol (iOS)
  //  * @example // (iOS)
  //  * image="plus"
  //  * @example // (Android)
  //  * image="ic_menu_add"
  //  * - TODO: Allow images other than those included in SF Symbol and resources drawables
  //  */
  // image?: string;
  // /**
  //  * (Android and iOS13+ only)
  //  * - The action's image color.
  //  */
  // imageColor?: number | ColorValue;
  /**
   * (Android and iOS14+ only)
   * - Actions to be displayed in the sub menu
   * - On Android it does not support nesting next sub menus in sub menu item
   */
  children?: MenuAction[];
  /**
   * Whether nested children should be inline (separated by divider) or nested (sub menu)
   */
  // displayChildrenInline?: boolean;
};
