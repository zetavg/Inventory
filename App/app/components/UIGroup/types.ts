import React from 'react';
import { Text, TextInput, TouchableWithoutFeedback, View } from 'react-native';
import { SymbolWeight } from 'react-native-sfsymbols';

import Icon from '../Icon';

type RenderFunctionProps = {
  textProps: {
    style?: React.ComponentProps<typeof Text>['style'] | undefined;
  };
  iconProps: {
    color?: string;
    size: number;
    sfSymbolWeight?: SymbolWeight;
    showBackground?: boolean;
    style?: React.ComponentProps<typeof Icon>['style'] | undefined;
  };
};

export type UIGroupProps = {
  children?: React.ReactNode;
  header?: string | JSX.Element;
  footer?:
    | string
    | ((context: {
        textProps: React.ComponentProps<typeof Text>;
        iconProps: Partial<React.ComponentProps<typeof Icon>>;
      }) => React.ReactNode);
  largeTitle?: boolean;
  headerRight?: string | JSX.Element;
  transparentBackground?: boolean;
  loading?: boolean;
  placeholder?: string;
  asSectionHeader?: boolean;
  asPlaceholderContent?: boolean;
} & React.ComponentProps<typeof View>;

export type UIGroupTitleButtonProps = {
  primary?: boolean;
  children?: string | ((context: RenderFunctionProps) => React.ReactNode);
} & Omit<React.ComponentProps<typeof TouchableWithoutFeedback>, 'children'>;

export type UIGroupFirstGroupSpacingProps = {
  iosLargeTitle?: boolean;
} & React.ComponentProps<typeof View>;

export type ListItemProps = {
  label?: string;
  labelTextStyle?: React.ComponentProps<typeof Text>['style'];
  detail?:
    | string
    | React.ReactNode
    | ((context: {
        textProps: React.ComponentProps<typeof Text>;
        iconProps: Partial<React.ComponentProps<typeof Icon>>;
      }) => React.ReactNode);
  detailTextStyle?: React.ComponentProps<typeof Text>['style'];
  monospaceDetail?: boolean;
  icon?:
    | React.ComponentProps<typeof Icon>['name']
    | ((p: {
        iconProps: Partial<React.ComponentProps<typeof Icon>>;
      }) => JSX.Element);
  iconColor?: React.ComponentProps<typeof Icon>['color'];
  rightElement?:
    | React.ReactNode
    | ((context: RenderFunctionProps) => React.ReactNode);
  verticalArrangedLargeTextIOS?: boolean;
  verticalArrangedNormalLabelIOS?: boolean;
  verticalArrangedIOS?: boolean;
  selected?: boolean;
  navigable?: boolean;
  onPress?: () => void;
  onLongPress?: () => void;
  button?: boolean;
  destructive?: boolean;
  disabled?: boolean;
  adjustsDetailFontSizeToFit?: boolean;
} & React.ComponentProps<typeof View>;

export type ListItemSeparatorProps = {
  leftInsetIOS?: number;
  color?: string;
  forItemWithIcon?: boolean;
};

export type ListItemRenderItemContainerProps = {
  children: React.ReactNode;
  isFirst?: boolean;
  isLast?: boolean;
};

export type ListTextInputItemProps = {
  label?: string;
  unit?: string;
  onUnitPress?: () => void;
  disabled?: boolean;
  readonly?: boolean;
  horizontalLabel?: boolean;
  controlElement?: React.ReactNode;
  rightElement?:
    | React.ReactNode
    | ((context: RenderFunctionProps) => React.ReactNode);
  monospaced?: boolean;
  small?: boolean;
  inputElement?:
    | React.ReactNode
    | ((context: RenderFunctionProps) => React.ReactNode);
} & React.ComponentProps<typeof TextInput>;

export type ListTextInputItemButtonProps = {
  children?: string | ((context: RenderFunctionProps) => React.ReactNode);
} & Omit<React.ComponentProps<typeof TouchableWithoutFeedback>, 'children'>;
