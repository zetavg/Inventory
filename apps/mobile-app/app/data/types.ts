export * from '@invt/data/types';
import {
  DataTypeName,
  DataTypeWithID,
  InvalidDataTypeWithID,
} from '@invt/data/types';

// Below is for backward compatibility.

export type DataTypeWithAdditionalInfo<T extends DataTypeName> =
  DataTypeWithID<T>;

export type InvalidDataTypeWithAdditionalInfo<T extends DataTypeName> =
  InvalidDataTypeWithID<T>;

export const DATA_ADDITIONAL_INFO_KEYS = [
  '__type',
  '__id',
  '__rev',
  '__deleted',
  '__created_at',
  '__updated_at',
] as const;
