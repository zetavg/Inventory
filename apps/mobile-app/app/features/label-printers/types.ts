import { z } from 'zod';

export const EnumOption = z.object({
  enum: z.array(z.string()),
  default: z.string().optional(),
  saveLastValue: z.boolean().optional(),
});

export const BooleanOption = z.object({
  type: z.literal('boolean'),
  default: z.boolean().optional(),
  saveLastValue: z.boolean().optional(),
});

export const StringOption = z.object({
  type: z.literal('string'),
  minLength: z.number().optional(),
  maxLength: z.number().optional(),
  default: z.string().optional(),
  choices: z.array(z.string()).optional(),
  saveLastValue: z.boolean().optional(),
});

export const IntegerOption = z.object({
  type: z.literal('integer'),
  minimum: z.number().optional(),
  maximum: z.number().optional(),
  default: z.number().optional(),
  choices: z.array(z.number()).optional(),
  saveLastValue: z.boolean().optional(),
});

export const Options = z.record(
  EnumOption.or(BooleanOption).or(StringOption).or(IntegerOption),
);
export type OptionsT = z.infer<typeof Options>;

export const Label = z.record(z.string());
export type LabelT = z.infer<typeof Label>;

export const PrinterConfig = z.object({
  options: Options,
  getLabel: z.function(),
  print: z.function(),
  getPreview: z.function().optional(),
});
export type PrinterConfigT = z.infer<typeof PrinterConfig>;
