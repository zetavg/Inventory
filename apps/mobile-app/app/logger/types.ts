export const LOG_LEVELS = [
  'debug',
  'info',
  'log',
  'success',
  'warn',
  'error',
] as const;

export type LogLevel = (typeof LOG_LEVELS)[number];

export type Log = {
  level: LogLevel;
  user?: string;
  module?: string;
  function?: string;
  message: string;
  details?: string;
  stack?: string;
  timestamp: number;
};
