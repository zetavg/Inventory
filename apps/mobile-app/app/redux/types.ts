import { Reducer } from 'redux';

import { DeepPartial } from '@app/utils/types';

export type PersistableReducer<R extends Reducer> = R extends Reducer<
  infer S
  // infer A,
  // infer PreloadedState
>
  ? R & {
      dehydrate?: (state: S) => any;
      rehydrate?: (dehydratedState: any) => DeepPartial<S>;
      dehydrateSensitive?: (state: S) => any;
      rehydrateSensitive?: (dehydratedState: any) => DeepPartial<S>;
    }
  : never;
