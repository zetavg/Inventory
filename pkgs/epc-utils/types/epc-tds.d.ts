declare module 'epc-tds' {
  /** No types yet. See: https://www.npmjs.com/package/epc-tds */
  const noTypesYet: any;
  export default noTypesYet;
  const Giai96 = noTypesYet;
  export { Giai96 };
}
