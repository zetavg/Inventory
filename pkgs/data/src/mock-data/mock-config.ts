import { ConfigType } from '../schema';
import { GetConfig } from '../types';

export const CONFIG: ConfigType = {
  uuid: 'mock-config-uuid',
  rfid_tag_company_prefix: '0000000',
  rfid_tag_individual_asset_reference_prefix: '100',
  rfid_tag_access_password: '12345678',
  default_use_mixed_rfid_tag_access_password: false,
  rfid_tag_access_password_encoding: '082a4c6e',
  collections_order: [],
};

export const getConfig: GetConfig = async () => CONFIG;
