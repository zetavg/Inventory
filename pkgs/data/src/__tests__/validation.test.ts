import { CONFIG, getConfig } from '../mock-data/mock-config';
import { DataTypeWithID } from '../types';
import getValidation, { ValidationError } from '../validation';

describe('ValidationError', () => {
  it('instances should also be instanceof Error', () => {
    const err = new ValidationError([]);
    expect(err instanceof Error).toBe(true);
  });
});

describe('validate', () => {
  describe('for item', () => {
    it('validates if collection_id is an ID of a existing collection', async () => {
      const { validate: validate } = getValidation({
        getConfig,
        getDatum: async (type, id) => {
          if (type === 'collection' && id === 'valid_collection_id') {
            const collection: DataTypeWithID<'collection'> = {
              __type: 'collection',
              __id: 'valid_collection_id',
              icon_name: '',
              icon_color: '',
              name: '',
              collection_reference_number: '1234',
              config_uuid: CONFIG.uuid,
              __valid: true,
            };
            return collection as any;
          }
          return null;
        },
        getData: async () => [],
        getRelated: async () => null,
      });

      const issues = await validate({
        __type: 'item',
        collection_id: 'no_such_collection',
        config_uuid: CONFIG.uuid,
      });

      expect(issues).toMatchSnapshot();

      const noIssues = await validate({
        __type: 'item',
        collection_id: 'valid_collection_id',
        config_uuid: CONFIG.uuid,
      });

      expect(noIssues).toStrictEqual([]);
    });
  });
});
