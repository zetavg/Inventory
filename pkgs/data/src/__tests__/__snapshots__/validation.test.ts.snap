// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate for item validates if collection_id is an ID of a existing collection 1`] = `
[
  {
    "code": "custom",
    "message": "Can't find collection with ID "no_such_collection"",
    "path": [
      "collection_id",
    ],
  },
]
`;
